### PR TITLE
Simplification: Drop `sendTransaction`

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

Here's what using `sendTransaction` looks like now:

Screenshot 2024-05-30 at 6 28 00 PM

I think we can make a few improvements!
- To wait for the transaction to complete devs need to:
    - Know to take the hash from the output and send a request to wait for it to complete.
    - Either:
        - Have a `viem` `PublicClient` already
        - Grab one from `wallet.getSigner("viem").publicClient` 
- We're using an `ethers` `Deferable<TransactionRequest>` as the input, that's confusing, especially considering most of the SDK uses `viem` based types. I've included a pic of the type below to illustrate that a non-crypto dev would be confused by the range of allowed options.

Screenshot 2024-05-30 at 6 28 34 PM

High level functions like `sendTransaction` should either drastically simplify something when compared to using the signer directly, or be removed.

This PR does a few things to simplify `sendTransaction`
    - Simplify and limit the parameters, advanced usage should be completed with `getSigner("viem")`
    - Within the function, wait for the transaction to complete.
    
Here's what sending an equivalent transaction would look like with the changes:

Screenshot 2024-05-30 at 7 22 07 PM

Or, we could delete this method entirely since using the signer alone is more ergonomic than the current method. @afeight wdyt, have a sense for if devs would fine the simplified `sendTransaction` useful?

**Edit: Decided to go with this approach**

Further justification [here](https://crossmint.slack.com/archives/C063Q6TSU9K/p1717174163275389?thread_ts=1717116025.184179&cid=C063Q6TSU9K)

Screenshot 2024-05-30 at 8 36 12 PM

This PR also clarifies and tidies a few pieces internally:
- Removes `ethers` as as dep, and deletes many newly unneeded pieces.
- Clarifies some Gelato bundler related edge cases. I've annotated this in the diff.

## Test plan

Locally on the demo site on branch `main-w3a` I:
- Updated the mint function to use the wallet client from `getSigner`.
- Confirmed
    - The mint works
    - Logs & performance tracking functionality are maintained

Passkey release QA that'll include:
- Testing a polygon CDK chain (Astar for ex.) 
- Wallet Connect

TS Compiling and Existing Tests

### CODE REVIEW

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 1]
```diff
@@ -117,33 +98,21 @@ export class EVMAAWallet extends LoggerWrapper {
...
-                    blobs: undefined,
-                    blobVersionedHashes: undefined,
-                    kzg: undefined,
-                    sidecars: undefined,
-                    type: undefined,
-                    chain: null,
+                    to: transaction.to,
+                    value: transaction.value,
+                    gas: transaction.gas,
+                    nonce: transaction.nonce,
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:110]: Do we need nonce here?

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 2]
```diff
@@ -171,7 +167,7 @@ export class EVMAAWallet extends LoggerWrapper {
                             abi: erc20,
                             functionName: "transfer",
                             args: [toAddress, (config as ERC20TransferType).amount],
-                            ...this.getLegacyTransactionFeesParamsIfApply(),
+                            ...(usesGelatoBundler(this.chain) && gelatoBundlerProperties),
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:164]: The repetition I introduce here is cleaned up in https://github.com/Crossmint/crossmint-sdk/pull/502

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 3]
```diff
@@ -253,28 +249,4 @@ export class EVMAAWallet extends LoggerWrapper {
...
-                ...(maxFeePerGas && { maxFeePerGas: BigInt(maxFeePerGas.toString()) }),
-                ...(maxPriorityFeePerGas && { maxPriorityFeePerGas: BigInt(maxPriorityFeePerGas.toString()) }),
-            };
-        } else {
-            if (maxFeePerGas || maxPriorityFeePerGas) {
-                console.warn(
-                    "maxFeePerGas and maxPriorityFeePerGas are not supported on this chain as it supports Legacy Transacitons. Ignoring them."
-                );
-            }
-            return {
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:272]: I could be missing something, but I think EIP1559 may have been conflated w/ the Gelato situation, coincidentally, this has worked fine so far because polygon CDK chains, which ZD uses Gelato for, don't support 1559.

File: packages/client/wallets/aa/src/utils/blockchain.ts [hunk 4]
```diff
@@ -14,55 +10,6 @@ export type VerifyMessageParams = {
...
 };
 
-/**
- * We need to support odd value and data fields in a tx, as that data comes from WC
- * ZeroDev implements hexlify, but doesn't support to send an odd string in value or data fields
- * so we need to make the hexlify fix manually when sending a transaction.
- * That fix can be found here:
- * https://github.com/ethers-io/ethers.js/commit/a12030ad29aa13c02aa75d9e0860f4986a0043b4#diff-047e7ebfdbd5c41e762cda03593bd15ed8b3121dece67262729dfcbde7040818R221
- * And more context on this issue:
- * https://github.com/ethers-io/ethers.js/issues/614
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/utils/blockchain.ts:24]: I suspect viem handles this

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 5]
```diff
@@ -117,38 +115,35 @@ export class EVMAAWallet extends LoggerWrapper {
         });
     }
 
-    //TODO @matias: review this method.
-    // First, I would like to use TransactionRequest from viem instead of ethers.
-    // Second, we need to check if chain supports eip-1559 ro not:
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:122]: If you don't send the params `kernelClient.sendTransaction` will figure it out and handle for you.

File: packages/client/wallets/aa/src/adapters/reservoir.ts [hunk 6]
```diff
@@ -42,6 +41,7 @@ export function reservoirAdapter(wallet: EVMAAWallet): ReservoirWallet {
                 ...(stepData.gas && {
                     gas: hexToBigInt(stepData.gas as any),
                 }),
+                ...(usesGelatoBundler(wallet.chain) && gelatoBundlerProperties),
```
Comment (devlyn37) [inline packages/client/wallets/aa/src/adapters/reservoir.ts:44]: The signer we expose will handle this soon

**Comment by alfonso-paella** [comment]
Is this a breaking change? If so, what's the plan for a responsible deprecation?

File: packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts
Comment (devlyn37) [inline packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts:1]: The changes here make sure we're not losing any functionality we had in `sendTransaction` (logging + performance metrics).

**Comment by devlyn37** [comment]
> Is this a breaking change? If so, what's the plan for a responsible deprecation?

I'll message Racino XD

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts (hunk 5 above)
Comment (devlyn37) [reply packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:110]: More extensive testing will be included the passkey release QA

File: packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts [hunk 7]
```diff
@@ -34,21 +34,34 @@ export function toCrossmintSmartAccountClient<Client extends SmartAccountClient<
...
+                        ...userOperation,
+                        ...(usesGelatoBundler(crossmintChain) && gelatoBundlerProperties),
+                    },
+                    middleware,
+                    account,
+                };
 
-            logInfo(`[CrossmintSmartWallet.sendUserOperation] - params: ${{ userOperation, middleware, account }}`);
-            return smartAccountClient.sendUserOperation({ userOperation, middleware, account });
+                logInfo(`[CrossmintSmartWallet.sendUserOperation] - params: ${params}`);
```
Comment (afeight) [inline packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts:62]: do we have the annotation for `@LogInputOutput` from matias?

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 8]
```diff
@@ -211,26 +173,4 @@ export class EVMAAWallet extends LoggerWrapper {
...
-            };
-        } else {
-            if (maxFeePerGas || maxPriorityFeePerGas) {
-                console.warn(
-                    "maxFeePerGas and maxPriorityFeePerGas are not supported on this chain as it supports Legacy Transacitons. Ignoring them."
-                );
-            }
-            return gelatoBundlerProperties;
-        }
-    }
```
Comment (afeight) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:235]: I forget if Matias or Maxi added this, there was some issue when we send transactions to the gelato bundler where we needed this
This is forced when we use astar because entry point v6 is the only one currently supported

**Review by afeight — APPROVED** [review]
I just checked logs, we're the only people that have used this function in the last 3 months
I'm happy to drop it

File: packages/client/wallets/aa/src/adapters/reservoir.ts [hunk 9]
```diff
@@ -29,8 +29,7 @@ export function reservoirAdapter(wallet: EVMAAWallet): ReservoirWallet {
         },
         handleSendTransactionStep: async (chainId, stepItem, _) => {
             const stepData = stepItem.data;
-            return await wallet.sendTransaction({
-                chainId: chainId,
```
Comment (afeight) [inline packages/client/wallets/aa/src/adapters/reservoir.ts:33]: have we tested this? We'll need to plug in to reservoir SDK

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts (hunk 8 above)
Comment (devlyn37) [reply packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:235]: `smartAccount.ts` has a replacement for this!

File: packages/client/wallets/aa/src/adapters/reservoir.ts (hunk 9 above)
Comment (devlyn37) [reply packages/client/wallets/aa/src/adapters/reservoir.ts:33]: No! To save time I'm waiting until the next release is ready, I'm keeping a tally of anything that needs manual testing and this is included. This'll also help inform plans for test automation

Comment (devlyn37) [reply packages/client/wallets/aa/src/adapters/reservoir.ts:33]: 🤠🤠🤠

File: packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts (hunk 7 above)
Comment (devlyn37) [reply packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts:62]: wdym by annotation here?

Using the `logInputOutput` function rn looks like:
```TS
export const getViemNetwork = logInputOutput((cmChain: EVMBlockchainIncludingTestnet) => {
// function contents here.
}, "getViemNetwork");
```

Comment (devlyn37) [reply packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts:62]: I'm going to merge this one @afeight can refine later if needed!
